include/vcmi/FactionMember.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * FactionMember.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#pragma once
+
+#include "Entity.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+class BonusList;
+
+namespace PrimarySkill
+{
+    enum PrimarySkill : int8_t;
+}
+
+class DLL_LINKAGE FactionMemberBase: public IConstBonusProvider, public INativeTerrainProvider
```

Comment:
No data in the class, then it's interface. You can name it in similar way IFactionMember

Reply:
@IvanSavenko this is my thought initially, what do you think?

Reply:
This is not "interface" since it also provides implementation of its methods

Reply:
For me, this is not principial - I can name it both way if we all agree.

Reply:
@IvanSavenko @Nordsoft91 which names are you prefer more for this non-data abstract classes? `XXXBase` or `IXXX`?

Reply:
Looking at our library code: `I` prefix is used only for classes that have all or almost all of their method as pure virtual.
Example: IBattleInfo / IBattleState
Abstact classes that have *some* of their methods implemented (and some are pure virtual) usually don't have `I` prefix - example: battle::Unit

Reply:
@IvanSavenko , this approach I would suggest to change. Virtual or stub methods - something important? Class is mostly about data. If no data provided - then it's interface for me.

Reply:
If we talk about "interface" as in 'interface' keyword in many languages then it should *not* provide implementation. In many languages it is actually illegal to have implementation inside interface.

For virtual/stub classes term "Abstract" is more common (but largely unused in vcmi codebase).

AbstactFactionMember / AFactionMember / FactionMemberBase are all reasonable names for me.

Reply:
@Nordsoft91 
Thinking about it a bit more: this is abstract class. A textbook abstract class. Not an interface.

- Interface class (sometimes - protocol) - describes a set of method signatures, but does not implements them.
- Abstract class - provides incomplete/partial implementation of an interface and can no be instantiated directly.
- Concrete class - final class that provides complete functionality.

Zero objections against using *some* naming scheme for abstract classes, but it should be different from interfaces. Personally XXXBase is most clear from my point of view, but any other form such as `A` prefix (for consistency with interfaces) is OK for me, as long as there is distinction between interfaces naming scheme and abstract classes.

Reply:
in the end, I believe it's up to @rilian-la-te to decide. All options I, A, Base suggested are ok, even initial one. Not really important

Reply:
Will use scheme with `AXXX`, will be appeared with next force-push.